### ISSUE TITLE
MariaDB4j on Java 21

### ISSUE THREAD

**Issue by fmartinou** [MariaDB4j/MariaDB4j#903]
Hi,

We use mariadb4j  from a while.
We recently are moving from java17 to java21 ans we now get errors at startup:

```
> Task :test

test > initializationError FAILED
    ch.vorburger.exec.ManagedProcessException at DB.java:179
        Caused by: ch.vorburger.exec.ManagedProcessException at ManagedProcess.java:310
            Caused by: org.apache.commons.exec.ExecuteException at DefaultExecutor.java:404
```

(we use the version `3.0.1`)

Can mariadb4j be used with java21?

**Comment by vorburger** [comment]
> Can mariadb4j be used with java21?

@fmartinou you may be the first one to try... 

But we should make it! Let's? I'm happy to review and merge PRs for this. (But I don't have time to fix it myself for you.)

The Error Message above is not very helpful - isn't there anything else afterwards? 

Would you be wiling to `git clone` this project and `./mvnw test` it under Java 21 and see what happens?

**Comment by vorburger** [comment]
> Would you be wiling to git clone this project and ./mvnw test it under Java 21 and see what happens?

On the current `main` branch, this failed with:

```
[INFO] --- compiler:3.13.0:compile (default-compile) @ mariaDB4j-core ---
[INFO] Recompiling the module because of changed source code.
[INFO] Compiling 10 source files with javac [debug target 17] to target/classes
[INFO] -------------------------------------------------------------
[WARNING] COMPILATION WARNING :
[INFO] -------------------------------------------------------------
[WARNING] system modules path not set in conjunction with -source 17
[INFO] 1 warning
[INFO] -------------------------------------------------------------
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] warnings found and -Werror specified
[INFO] 1 error
```

But I've fixed that with #917, based on e.g. https://github.com/quarkusio/quarkus/issues/23799 and https://stackoverflow.com/q/43102787/421602, see also https://openjdk.org/jeps/247.

With that, the build of `main` works and tests pass on Java 21.

**Comment by fmartinou** [comment]
Thank you very much!! 🥳
